internal/kafka/internal/services/clusters.go

```diff
@@ -665,3 +668,120 @@ func (c clusterService) IsStrimziKafkaVersionAvailableInCluster(cluster *api.Clu
 	}
 	return false, nil
 }
+
+type KafkaStreamingUnitCountPerCluster struct {
+	Region        string
+	InstanceType  string
+	ClusterId     string
+	Count         int32
+	CloudProvider string
+	MaxUnits      int32
+}
+
+func (KafkaStreamingUnitCountPerCluster KafkaStreamingUnitCountPerCluster) isSame(kafkaPerRegionFromDB *KafkaPerClusterCount) bool {
+	return KafkaStreamingUnitCountPerCluster.CloudProvider == kafkaPerRegionFromDB.CloudProvider &&
+		KafkaStreamingUnitCountPerCluster.ClusterId == kafkaPerRegionFromDB.ClusterId &&
+		KafkaStreamingUnitCountPerCluster.InstanceType == kafkaPerRegionFromDB.InstanceType &&
+		KafkaStreamingUnitCountPerCluster.Region == kafkaPerRegionFromDB.Region
+}
+
+type KafkaStreamingUnitCountPerClusterList []KafkaStreamingUnitCountPerCluster
+
+func (kafkaStreamingUnitCountPerClusterList KafkaStreamingUnitCountPerClusterList) GetStreamingUnitCountForClusterAndInstanceType(clusterId, instanceType string) int {
+	for _, KafkaStreamingUnitCountPerCluster := range kafkaStreamingUnitCountPerClusterList {
+		if KafkaStreamingUnitCountPerCluster.ClusterId == clusterId && KafkaStreamingUnitCountPerCluster.InstanceType == instanceType {
+			return int(KafkaStreamingUnitCountPerCluster.Count)
+		}
+	}
+
+	return 0
+}
+
+// KafkaPerClusterCount is a struct used to query the database using a "group by" clause
+type KafkaPerClusterCount struct {
+	Region        string
+	InstanceType  string
+	ClusterId     string
+	Count         int32
+	CloudProvider string
+	SizeId        string
+}
+
+type ClusterSelection struct {
+	CloudProvider         string
+	ClusterID             string
+	Region                string
+	SupportedInstanceType string
+	DynamicCapacityInfo   api.JSON
+}
+
+func (c *clusterService) FindStreamingUnitCountByClusterAndInstanceType() (KafkaStreamingUnitCountPerClusterList, error) {
+
+	var clusters []*ClusterSelection
+	dbConn := c.connectionFactory.New().
+		Model(&api.Cluster{}).
+		Where("cluster_id != ''")
+
+	if err := dbConn.Scan(&clusters).Error; err != nil {
+		return nil, apiErrors.NewWithCause(apiErrors.ErrorGeneral, err, "failed to list clusters")
+	}
+
+	streamingUnitsCountPerRegion := KafkaStreamingUnitCountPerClusterList{}
+
+	// pre-populate regions count with zero count values.
+	// This is useful and it ensures that the count drops to 0 for each instance type and region when all the kafkas are removed
+	for _, clusterSelection := range clusters {
+		supportedInstanceTypes := strings.Split(clusterSelection.SupportedInstanceType, ",")
+
+		c := &api.Cluster{DynamicCapacityInfo: clusterSelection.DynamicCapacityInfo}
+		clusterDynamicCapacityInfo, _ := c.RetrieveDynamicCapacityInfo()
```

Comment:
The error shouldn't be ignored

Reply:
An error won't occur here. 
We'll be parsing a valid json at this point as we'd have stored a valid json.
The `RetrieveDynamicCapacityInfo()` method could also be made to not return the error at all.

Reply:
+1, I'd agree on that.

Reply:
+1 agree